### ISSUE TITLE
KEES Carrier causes MASSIVE fps drop upon KAS attachment

### ISSUE THREAD

**Issue by StarkRG** [N3h3miah/OrbitalMaterialScience#122]
After attaching the carrier to a vessel my FPS drops to around 1-0.5 frames per MINUTE when I'm on a ladder, or about 1 FPS while floating free. If I can manage to get back inside the vessel the FPS returns to normal and remains that way even on subsequent EVAs.

It only seems to be the KEES carrier that does this, other items, including the KEES experiments work fine. I'm going to be removing the KASModuleGrab from the carrier until this can be fixed, but their large size makes them (and the associated experiments) almost useless until I can research cargo bays.

EDIT: Whoops, meant to say it's the Passive Experiment Carrier, not the Payload Carrier. Perhaps a solution would be to combine the two and have the experiment slide out when needed (thus also eliminating the requirement for KAS).

**Comment by N3h3miah** [comment]
This sounds like #115 can you check the debug console for "NullReferenceException: Object reference not set to an instance of an object"  thanks

**Comment by StarkRG** [comment]
Hmm, I'm not getting that, but I just noticed the same issue while attaching something not from OMS, so perhaps it's a KAS issue. The strange thing is that I didn't notice this issue on a previous vessel.

**Comment by N3h3miah** [comment]
I label this "not this mod" for the time being. please let me know when you could track it down to either KAS, this mod or something else. thanks.

**Comment by mwerle** [comment]
Confirm, although I've only noticed if the PEC is detached from the vessel and floating as debris.  Haven't seen this just when attaching/detaching normally, but will test further.
No corresponding log output (Player or KSP).

What is possibly happening is that either KAS or OMS (or another mod??) is performing excessive processing in an OnUpdate() event.  N3h3miah - I encountered this occasionally before you refactored the code and moved the checkForExp() call into onUpdate(), so I don't think that triggered this.

I've not been able to reproduce reliably, so haven't raised this before.
